### DISCUSSION TITLE
add another class during svg rendering

### DISCUSSION THREAD

**Discussion by gerardoscoppetta** [opensheetmusicdisplay/opensheetmusicdisplay#1532] [Q&A] [answered]
HI.
When rendering as SVG is it possible to have the vf-notehead classes add another class that specifies whether it is a musical note or a rest?
Thank you.

**Comment by sschmidTU** [comment]
Hi, why do you need that? You know that you can check note.isRest() or note.sourceNote.isRest(), right?
This would only be useful if you looked only at the SVG, without using OSMD to get SVG nodes.

**Reply by gerardoscoppetta** [reply]
HI. yes, once printed from osmd I would like to modify the svg by adding other small elements corresponding to the notes and acting on the notes themselves for example by coloring the notes.
Here I'll post a test page where I'm doing some tests.
With just one pentagram I think I can do everything without problems. However, when I use two voices as in the example it is more difficult to obtain the exact references. it would be perfect if you could identify the notes and rests and assign unique classes and also assign a class that identifies the left hand staff and right hand staff as in the example in the link I give you below.
Thanks so much.
https://www.suonosubito.net/test_opensheetmusicdisplay/test_piano_color_note.php

**Reply by sschmidTU** [reply]
Again, you can color notes without needing SVG classes, see here:
https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/wiki/Exploring-the-Demo#coloring-notes-without-re-rendering

I don't see how you need classes for rests or each staff (piano left/right hand) when you have all the necessary information in OSMD. You can also check whether a note is from the right or left hand in OSMD.
But if you can explain it, I might be convinced and we can add more classes.

**Reply by sschmidTU** [reply]
by the way, we just added a few SVG classes recently, see issue https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/issues/1531, it's a related discussion. Just in case you or someone else reading this haven't noticed

**Reply by gerardoscoppetta** [reply]
ok. I'll try to convince you :)
the svg file is very versatile. The idea is to be able to insert everything that comes to mind under the notes. It could be very creative from an educational point of view. To do this, it would therefore be essential for a staff to have first of all a class on the node to identify the voices of the staffs (Part 1, Part 2 etc if of two or more voices) and then in a sub-node of each voice a class to identify the notes so as not to be confused with rests and a class that identifies rests so as not to be confused with notes.
If it is possible I invite you to do it because it would be a great added value.
Thank you very much.

**Reply by sschmidTU** [reply]
So you want to get all information purely from the SVG file and ignore all information the OSMD object could have?
Because as I said, you can get all this information from OSMD.

**Reply by gerardoscoppetta** [reply]
So maybe I'm not able to figure out how to do it.
For example, if I wanted to print a score where under all the notes I would like to insert a small box with a different icon for each note, would you say the same would be possible using the OSMD references?

**Reply by sschmidTU** [reply, marked as answer]
You can get all this information from the OSMD object, including the SVG node.
There are many ways you can get note objects, like this:
`osmd.graphic.MeasureList[0][0].staffEntries[0].graphicalVoiceEntries[0].notes`
from there, you can get the SVG node:
`osmd.graphic.MeasureList[0][0].staffEntries[0].graphicalVoiceEntries[0].notes[0].getSVGGElement()`
(MeasureList[0] contains all vertical measures at index 0 that have the same measure number. For piano, there will be 2 vertical measures in it.)
You can also check whether it's a rest:
`osmd.graphic.MeasureList[0][0].staffEntries[0].graphicalVoiceEntries[0].notes[0].sourceNote.isRest()`

There would be no end to how many extra classes we could write into the SVG besides whether it's a rest, like whether it has a sharp accidental and so on, it would bloat the SVG and degrade performance.
So the better approach is to get the OSMD object representing the SVG node, as I described.
And by the way, rest notes are also (musical) notes ;) There is even a piece including only rest notes (Cage's 4'33).
